coffea/processor/executor.py

```diff
@@ -555,30 +623,152 @@ def __call__(
         if self.compression is not None:
             function = _compression_wrapper(self.compression, function)
 
-        def processwith(pool):
-            gen = _futures_handler(
-                {pool.submit(function, item) for item in items}, self.tailtimeout
+        def processwith(pool, mergepool):
+            reducer = _reduce(self.compression)
+            FH = FuturesHolder(
+                set(pool.submit(function, item) for item in items), refresh=2
             )
+
+            nparts, minred, maxred = 5, 4, 100
+            if isinstance(self.merging, tuple):
+                if len(self.merging) == 3:
+                    nparts, minred, maxred = self.merging
+
             try:
-                return accumulate(
-                    tqdm(
-                        gen if self.compression is None else map(_decompress, gen),
-                        disable=not self.status,
-                        unit=self.unit,
-                        total=len(items),
-                        desc=self.desc,
-                    ),
-                    accumulator,
-                )
-            finally:
-                gen.close()
+                with rich_bar() as progress:
+                    p_id = progress.add_task(
+                        self.desc, total=len(items), unit=self.unit
+                    )
+                    _mdesc = "Merging" if self.merging else "Merging (local)"
+                    p_id_m = progress.add_task(_mdesc, total=0, unit="merges")
+
+                    merged = None
+                    while len(FH.futures) + len(FH.merges) > 0:
+                        FH.update()
+                        merge_size = min(
+                            maxred, max(len(FH.completed) // nparts + 1, minred)
+                        )
+                        progress.update(
+                            p_id,
+                            advance=FH.done["futures"]
+                            - progress._tasks[p_id].completed,
```

Comment:
Is there a public method on the rich progress bar?

Reply:
there isn't but there's a different way to advance the progress